File: backend/src/main/java/com/grouply/backend/invitation/InvitationDTO.java
package com.grouply.backend.invitation;

import lombok.*;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class InvitationDTO {
    private Long id;
    private Long senderId;
    private Long recipientId;
    private Long projectId;
    private String status;
    private LocalDateTime createdAt;
}

File: frontend/src/dtos/InvitationDTO.ts
import type { InvitationStatus } from "../models/enums/InvitationStatus"

export class InvitationDTO {
    id: number
    senderId: number
    recipientId: number
    projectId: number
    status: InvitationStatus
    createdAt: string

    constructor(id: number, senderId: number, recipientId: number, projectId: number, status: InvitationStatus, createdAt: string) {
        this.id = id
        this.senderId = senderId
        this.recipientId = recipientId
        this.projectId = projectId
        this.status = status
        this.createdAt = createdAt
    }
}